src/wallet/wallet.cpp

```diff
@@ -704,18 +704,24 @@ void CWallet::SetUsedDestinationState(WalletBatch& batch, const uint256& hash, u
 {
     AssertLockHeld(cs_wallet);
     const CWalletTx* srctx = GetWalletTx(hash);
+    std::set<CTxDestination> tx_destinations;
```

Comment:
This must be an argument, like `std::set<CTxDestination>& tx_destinations`. In the call site you do
```cpp
    if (IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE)) {
        // Mark used destinations
        std::set<CTxDestination> tx_destinations;
        for (const CTxIn& txin : wtxIn.tx->vin) {
            const COutPoint& op = txin.prevout;
            SetUsedDestinationState(batch, op.hash, op.n, true, tx_destinations);
        }
        MarkDestinationsDirty(tx_destinations);
    }
```

Reply:
fixed